apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -193,8 +222,10 @@ int recovery_file(char *src, char* backup, char* backup_dir, int bufsize, int id
 	int fd1;
 	int fd2;
 	int ret;
+	struct stat st;
 	off_t offset;
 	off_t start_offset;
+	int filesize;
 	
 	/* If idx is ERROR, it means it didn't create src file so bring backup file */
 	printf("recovery_file src : %s idx : %d\n", src, idx);
```

Comment:
Line 232 `if (idx == ERROR || idx == 0 || idx == TEST_WRITECOUNT)`

if (idx == 0) -> File has no valid data, AND file has been truncated to length 0.

I get that we want to test shrink function, but
instead of unlinking the source file and renaming backup file,
We can also append to this shrinked file right?

Reply:
Also,
is recovery_file ever called with `idx == TEST_WRITECOUNT`

Reply:
Updated totally. hence resolve ths